xcp_d/workflows/bold/concatenation.py

```diff
@@ -501,3 +507,107 @@ def _combine_name(in_files):
         names.append(new_file)
 
     return names
+
+
+def _combine_concat_name(in_files):
+    """Remove unmatched entities from a list of files to produce a single name, plus dir and run.
+
+    Parameters
+    ----------
+    in_files : :obj:`list` of :obj:`list` of :obj:`str` or :obj:`list` of :obj:`str`
+        List of lists of file paths. Each sublist contains files for a single atlas.
+
+    Returns
+    -------
+    names : :obj:`list` of :obj:`str` or :obj:`str`
+        Combined names, containing only the entities that are present in all files in a sublist.
+        Also drops dir and run entities.
+
+    Examples
+    --------
+    >>> _combine_concat_name([
+    ...     ['/path/to/sub-01_task-rest_run-1_bold.nii.gz',
+    ...      '/path/to/sub-01_task-rest_run-2_bold.nii.gz'],
+    ... ])
+    ['/path/to/sub-01_task-rest_bold.nii.gz']
+
+    >>> _combine_concat_name([
+    ...     ['/path/to/sub-01_task-rest_dir-AP_run-1_bold.nii.gz',
+    ...      '/path/to/sub-01_task-rest_dir-PA_run-2_bold.nii.gz'],
+    ... ])
+    ['/path/to/sub-01_task-rest_bold.nii.gz']
+
+    >>> _combine_concat_name([
```

Comment:
Are you sure you don't want dir-AP here

Reply:
We concatenate across the run and dir entities because of HCP. It may not always be the behavior people want, but I think folks will want to treat data acquired across multiple directions as the same more often than not.